src/OfflineQuery.js

```diff
@@ -513,9 +513,23 @@ function matchesKeyConstraints(className, object, objects, key, constraints) {
       return true;
     }
     case '$geoWithin': {
-      const points = compareTo.$polygon.map(geoPoint => [geoPoint.latitude, geoPoint.longitude]);
-      const polygon = new ParsePolygon(points);
-      return polygon.containsPoint(object[key]);
+      if (compareTo.$polygon) {
+        const points = compareTo.$polygon.map(geoPoint => [
+          geoPoint.latitude,
+          geoPoint.longitude,
+        ]);
+        const polygon = new ParsePolygon(points);
+        return polygon.containsPoint(object[key]);
+      }
+      // $centerSphere
+      const [WGS84Point, maxDistance] = compareTo.$centerSphere;
```

Comment:
Would it make sense to also wrap this into an if statement? Just like `if (compareTo.$polygon) {`

Reply:
Done